### PR TITLE
Till report

### PR DESCRIPTION
**Pull request by kentpmckinney**
Adds the printing of till reports when a till is closed

### CODE REVIEW

File: src/main/resources/templates/utility/tillsession.html [hunk 1]
```diff
@@ -35,6 +35,8 @@
                     </tr>
                 </table>
                 <br>
+                <div class="alert alert-info"><span class="badge badge-primary">NOTE:</span> Closing the till will print a report to: <span class="badge badge-info" th:text="${reportPrinterName}"></span></div>
```
Comment (jashort) [inline src/main/resources/templates/utility/tillsession.html:38]: I like this message. I suggest adding a th:if= to test for `reportPrinterName == null or reportPrinterName.isBlank()`, and if it is null warn that the till report will not print automatically.

File: src/main/java/org/kumoricon/registration/utility/TillSessionController.java [hunk 2]
```diff
@@ -79,8 +79,10 @@ public String closeTill(Model model,
             }
         } else if (action.equals("Close Till")) {
             tillSessionService.closeSessionForUser(currentUser, tillName);
-
-            return "redirect:/utility/till?msg=Till%20Session%20Closed";
+            String reportPrinterName = settingsService.getCurrentSettings().getReportPrinterName();
```
Comment (jashort) [inline src/main/java/org/kumoricon/registration/utility/TillSessionController.java:93]: The way it stands now, if the report fails to print it will just throw an error message (for example: "Error: Printer named '' does not exist and no default printer set"). 
There's not really anything for regular users to do at that point (and no way for them to reprint anyway). 

If it fails, I think I'd like it to just redirect the user to the next page with an error message ("warning: your report did not print automatically" or something like that.) but let them continue with whatever they were doing. We can always print the report on demand from the back anyway. But getting the person going off shift off the computer and whoever should be going on shift in the seat checking people in is most important

File: src/main/java/org/kumoricon/registration/print/ReportPrintService.java [hunk 3]
```diff
@@ -0,0 +1,132 @@
+package org.kumoricon.registration.print;
```
Comment (jashort) [inline src/main/java/org/kumoricon/registration/print/ReportPrintService.java:1]: I like this. If you don't want to do it, I'm going to add a "Download" button that uses this create a downloadable PDF to /admin/tillsessions
(If you do want to do it, feel free. :) )

File: src/main/resources/templates/utility/tillsession.html (hunk 1 above)
Comment (kentpmckinney) [reply src/main/resources/templates/utility/tillsession.html]: Added to the PR

File: src/main/java/org/kumoricon/registration/utility/TillSessionController.java (hunk 2 above)
Comment (kentpmckinney) [reply src/main/java/org/kumoricon/registration/utility/TillSessionController.java:93]: Added to the PR

File: src/main/java/org/kumoricon/registration/print/ReportPrintService.java (hunk 3 above)
Comment (kentpmckinney) [reply src/main/java/org/kumoricon/registration/print/ReportPrintService.java:1]: Added to the PR
